vnext/IntegrationTests/TestRunner.cpp

```diff
@@ -52,76 +54,81 @@ TestResult TestRunner::RunTest(
+        false; // WebSocketJSExecutor can't register native log hooks.
+    devSettings->debugHost = "localhost:8081";
+    devSettings->liveReloadCallback = []() {}; // Enables ChakraExecutor
+    devSettings->errorCallback = [&result](string message) {
+      result.Message = Microsoft::Common::Unicode::Utf8ToUtf16(message);
+      result.Status = TestStatus::Failed;
+    };
+    devSettings->loggingCallback = std::move(loggingCallback);
+
+    // React instance scope
+    {
+      shared_ptr<ITestInstance> instance = GetInstance(
+          std::move(bundlePath), std::move(modules), std::move(devSettings));
+
+      InitializeLogging([&result, &functionCalled](
+                            RCTLogLevel logLevel, const char *message) {
+        if (TestStatus::Pending != result.Status)
+          return;
+
+        switch (logLevel) {
+          case RCTLogLevel::Error:
+          case RCTLogLevel::Fatal:
+            result.Message = Microsoft::Common::Unicode::Utf8ToUtf16(message);
+            result.Status = TestStatus::Failed;
+            ::SetEvent(functionCalled);
+            break;
+
+          case RCTLogLevel::Info:
+          case RCTLogLevel::Trace:
+          case RCTLogLevel::Warning:
+          default:
+            break;
+        }
+      });
+
+      instance->AttachMeasuredRootView(move(appName));
+
+      // Failure on instance creation.
+      if (TestStatus::Failed == result.Status)
+        return result;
+
+      AwaitEvent(functionCalled, result);
+
+      instance->DetachRootView();
+    }
+
+    // Tear down
+    CloseHandle(functionCalled);
+    functionCalled = NULL;
+  } catch (...) {
+    result.Message = Microsoft::Common::Unicode::Utf8ToUtf16(
+        boost::current_exception_diagnostic_information(/*verbose*/ true));
     result.Status = TestStatus::Failed;
-  };
-  devSettings->loggingCallback = std::move(loggingCallback);
-
-  // React instance scope
-  {
-    shared_ptr<ITestInstance> instance = GetInstance(
-        std::move(bundlePath), std::move(modules), std::move(devSettings));
-
-    InitializeLogging(
-        [&result, &functionCalled](RCTLogLevel logLevel, const char *message) {
-          if (TestStatus::Pending != result.Status)
-            return;
-
-          switch (logLevel) {
-            case RCTLogLevel::Error:
-            case RCTLogLevel::Fatal:
-              result.Message = Microsoft::Common::Unicode::Utf8ToUtf16(message);
-              result.Status = TestStatus::Failed;
-              ::SetEvent(functionCalled);
-              break;
-
-            case RCTLogLevel::Info:
-            case RCTLogLevel::Trace:
-            case RCTLogLevel::Warning:
-            default:
-              break;
-          }
-        });
-
-    instance->AttachMeasuredRootView(move(appName));
-
-    // Failure on instance creation.
-    if (TestStatus::Failed == result.Status)
-      return result;
-
-    AwaitEvent(functionCalled, result);
-
-    instance->DetachRootView();
   }
```

Comment:
we wouldn't get a stack trace here though? can we capture the stack trace from the exception?

Reply:
Stack trace is not what I target yet.
exception is just suspicious which may contribute the timeout. Here catching all exception is just help to narrow down the problem.
If I see this error, I will provide more logs like catch(std::exception) and log the stack.

---
In reply to: [337866067](https://github.com/microsoft/react-native-windows/pull/3492#discussion_r337866067) [](ancestors = 337866067)